### PR TITLE
[AMORO-3195]: OptimizerController decoupling to OptimizerGroupController

### LINKED ISSUE #3195: [Improvement]: OptimizerController decoupling to OptimizerGroupController
### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

OptimizerController now have OptimizerGroup method,We hope to clarify the responsibilities of each component. When using external containers, we do not need to additionally host them on Amoro Optimizer. 

### How should we improve?

- move OptimizerController optimizerGroup Methods to OptimizerGroupController
- web ui role to OptimizerController and OptimizerGroupController

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by mansonliwh**
## Why are the changes needed?

Close #3195 

## Brief change log

1. The backend project has split the OptimizerController into OptimizerController and OptimizerGroupController. The OptimizerController now contains the create and release methods, while the OptimizerGroupController retains all the original logic of the OptimizerController.
2. The frontend project has added a "Create Optimizer" button on the Optimizer page.

## How was this patch tested?

image

Run test locally before making this pull request

### CODE REVIEW

File: amoro-web/src/views/resource/index.vue [hunk 1]
```diff
@@ -92,6 +98,17 @@ export default defineComponent({
       state.showGroupModal = true
     }
 
+    const createOptimizer = (editRecord: any | null) => {
+      console.log('createOptimizer function called') // 添加日志
```
Comment (czy006) [inline amoro-web/src/views/resource/index.vue:102]: can be remove console

File: amoro-web/src/views/resource/components/CreateOptimizerModal.vue [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      })
+}
+
+function handleCancel() {
+  formRef.value.resetFields()
+  emit('cancel')
+}
+
+onMounted(() => {
+  console.log('Component mounted') // 添加日志
```
Comment (czy006) [inline amoro-web/src/views/resource/components/CreateOptimizerModal.vue:81]: can be remove console

**Comment by czy006** [comment]
image

**Review by tcodehuber — APPROVED** [review]
LGTM

**Comment by tcodehuber** [comment]
> image
> 
> Thank you for your contribution, do we still need to keep the Scala out button? cc @majin1102

I think we can remove this operation from `Optimizer Groups` tab.

**Review by czy006 — APPROVED** [review]
LGTM

**Review by majin1102 — APPROVED** [review]
LGTM

**Pull request merged**